src/docs/development/tools/devtools/code-size.md

```diff
@@ -0,0 +1,194 @@
+
+#### Using the dominator tree
+
+![Screenshot of code size dominator tree]({% asset tools/devtools/code_size_dominator_tree.png @path %}){:width="100%"}
+
+A dominator tree is a tree where each node's children are those nodes it
+immediately dominates. A node `d` dominates a node `n` if every entry point to
+`n` must go through `d`.
+
+To put it in context of app size analysis, imagine `package:a` imports both
+`package:b` and `package:c`, and both `package:b` and `package:c` import
+`package:d`.
+
+```
+package:a
+|__ package:b
+|   |__ package:d
+|__ package:c
+    |__ package:d
+```
+
+In this example, `package:a` dominates `package:d`, so the dominator tree for
+this data would look like:
+
+```
+package:a
+|__ package:b
+|__ package:c
+|__ package:d
+```
+
+This information is helpful for understanding why certain pieces of code are
+present into your compiled application. For example, if you are analyzing your
+app size and find an unexpected package included in your compiled app, you can
+use the dominator tree to trace the package to its root source.
+
+#### Using the call graph
+
+![Screenshot of code size call graph]({% asset tools/devtools/code_size_call_graph.png @path %}){:width="100%"}
+
+A call graph, unlike the dominator tree, shows the many-to-many relationships
+that exist between nodes of code size data. We'll again use this example:
+
+```
+package:a
+|__ package:b
+|   |__ package:d
+|__ package:c
+    |__ package:d
+```
+
+The call graph for this data would link `package:d` to its direct callers,
+`package:b` and `package:c`, instead of its "dominator", `package:a`.
+
+```
+package:a --> package:b -->
+                              package:d
+package:a --> package:c -->
+```
+
+## Diff Tab
+
+![Screenshot of code size diff]({% asset tools/devtools/code_size_diff.png @path %}){:width="100%"}
+
+### Loading code size files
+
+When you open the Diff tab, you'll see instructions to load "old" and "new" code
+size files. These files should be generated from two different versions of the
+same app, for example, the bits before and after a PR. Drag and drop these files
+in to their respective locations, and click "Analyze Diff".
+
+![Screenshot of code size diff loading screen]({% asset tools/devtools/code_size_load_diff.png @path %}){:width="100%"}
+
+See [Generating size files][] below for information on generating these files.
+
+### Treemap and table
+
+In the diff view, the treemap and tree table show only data that differs between
+the two imported size files.
+
+For questions about using the treemap, see [How to use the treemap][] below.
+
+## How to use the treemap
+
+A treemap is a visualization for hierarchical data. The space is broken up into
+rectangles, where each rectangle is sized and ordered by some quantitative
+variable (in this case, size in bytes). Inside of each rectangle (call one A),
+there are additional rectangles that exist one level deeper in the data
+hierarchy (children of A).
+
```

Comment:
Just to make it super explicit, say that the areas of the rectangles are proportional to the size they occupy in the application.